.evergreen/load-balancer.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/sh
+
+set -o errexit  # Exit the script with error if any of the commands fail
+
+start() {
+  echo "Starting HAProxy..."
+
+  cat <<EOF_HAPROXY_CONFIG >> $DRIVERS_TOOLS/haproxy.conf
+  defaults
+      mode tcp
+      timeout connect 5000ms
+      timeout client 5000ms
+      timeout server 5000ms
+
+  frontend mongos_frontend
+      bind *:8000
+      use_backend mongos_backend
+
+  frontend mongoses_frontend
+      bind *:8001
+      use_backend mongoses_backend
+
+  backend mongos_backend
+      mode tcp
+      server mongos 127.0.0.1:27017 check
+
+  backend mongoses_backend
+      mode tcp
+      server mongos_one 127.0.0.1:27017 check
+      server mongos_two 127.0.0.1:27018 check
+EOF_HAPROXY_CONFIG
+
+  cat $DRIVERS_TOOLS/haproxy.conf
+
+  /usr/sbin/haproxy -D -f $DRIVERS_TOOLS/haproxy.conf -p ./haproxy.pid
+
+  ADDRESS=$(curl -s http://169.254.169.254/latest/meta-data/public-hostname)
+  SINGLE_MONGOS_LB_URI="$ADDRESS:8000"
```

Comment:
Related to our offline conversation yesterday: if we expect the underlying sharded cluster to be started with auth/TLS, we either need to put credentials in this URI or otherwise instruct drivers to do so in their test runners. The former is simpler but definitely couples the LB to the cluster because we'd be making assumptions about how the cluster is set up. I'm not sure if there's prior art to go off of here. CC @jyemin maybe you have some thoughts about this?

Reply:
I agree that this URI should contain credentials, as well as `?loadBalanced=true`.

Also, are we planning to run these tests with TLS enabled?  If so we'll have to account for that as well.

Reply:
Yes, please add the required arguments to the connection string. This avoids ugly workarounds like we're seeing for mongo-orchestration (which doesn't yet expose tls options when required). Ideally, `mongo $SINGLE_MONGOS_LB_URI` should be able to connect and behave as expected.

Reply:
I think it's hard to see how to use this without integration into the Evergreen configuration file.  Can you add a variant that uses this file to start a sharded cluster (with and without auth), a proxy, and runs a test?

Reply:
How would this work if we're testing both with and without auth? Will we have to output two sets of env vars (e.g. `SINGLE_MONGOS_LB_URI` and `SINGLE_MONGOS_LB_AUTH_URI` as well as a similar set for `MULTI_*`) and have the driver pick which ones get plumbed down to the test runner in the Evergreen YAML config?

Reply:
I've updated to include auth and potential parameters, and the Evergreen config reference is here: https://github.com/mongodb/node-mongodb-native/pull/2787